File: Interview_Prep_App/native_java/javatemplate/src/main/java/com/tc2r/javatemplate/ui/main/SplashScreenActivity.java
package com.tc2r.javatemplate.ui.main;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import com.tc2r.javatemplate.R;

public class SplashScreenActivity extends AppCompatActivity {

    long screenTimer = 2000L;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);
        new Handler().postDelayed(() -> {
            Intent i = new Intent(SplashScreenActivity.this, MainActivity.class);
            startActivity(i);
            finish();
        }, screenTimer);
    }
}

File: Interview_Prep_App/native_kotlin/base/src/main/java/com/nudennie/interviewprepapp/ui/main/SplashScreenActivity.kt
package com.nudennie.interviewprepapp.ui.main

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import com.nudennie.interviewprepapp.R

class SplashScreenActivity : AppCompatActivity() {

    // This holds the time in milisec for the splash screen display
    val screenTimer = 4000L
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)
        Handler().postDelayed(Runnable { // This method executes once timer has ended



            // Starting app main activity
            val i = Intent(this@SplashScreenActivity, MainActivity::class.java)
            startActivity(i)

            // Closing this activity
            finish()

        }, screenTimer)
    }
}
